### ISSUE TITLE
Behavior of guidesEnabledProperty and twoFPointsEnabledProperty

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics#365]
Here's an interesting PhET-iO design question.  These 2 global Properties are initialized using query parameters. The enable the "Guides" and "2F Points" features:

`geometricOptics.global.view.options.guidesEnabledProperty`
`geometricOptics.global.view.options.twoFPointsEnabledProperty`

I'll discuss `guidesEnabledProperty`, but the same behavor is similar for both Properties.  If `guidesEnabledProperty` is set to `true`, it add a "Guides" checkbox to the control panel, which you can then use to show/hide guides.  If you set it to `false`, it hides the "Guides" feature AND any guides that were visible -- i.e., it disables the entire feature.  So it's not currently possible to (for example) show the guides, then hide the "Guides" checkbox. 

Should something change here, or is this OK? 

We could change so that these Properties (and query parameters) only show/hide the checkboxes in the control panels. But then we should change the names of the Properties and query parameters. Probably something like `showGuidesCheckboxProperty` and `show2FPointsCheckboxProperty`, with query parameters `showGuidesCheckbox` and `show2FPointsCheckbox`. (Hmmm... I like those names quite a bit more.)

**Comment by pixelzoom** [comment]
> We could change so that these Properties (and query parameters) only show/hide the checkboxes in the control panels. But then we should change the names of the Properties and query parameters. Probably something like `showGuidesCheckboxProperty` and `show2FPointsCheckboxProperty`, with query parameters `showGuidesCheckbox` and `show2FPointsCheckbox`. (Hmmm... I like those names quite a bit more.) 

The more I think about it, the more I'm convinces that this is what we want to do.

**Comment by arouinfar** [comment]
Great question @pixelzoom. 

>I'll discuss `guidesEnabledProperty`, but the same behavor is similar for both Properties.  If `guidesEnabledProperty` is set to `true`, it add a "Guides" checkbox to the control panel, which you can then use to show/hide guides.  If you set it to `false`, it hides the "Guides" feature AND any guides that were visible -- i.e., it disables the entire feature.  So it's not currently possible to (for example) show the guides, then hide the "Guides" checkbox. 
>
>Should something change here, or is this OK? 

This sounds problematic to me. Is there a way to disentangle things so that the feature can be turned on, but the checkbox can be hidden?

> We could change so that these Properties (and query parameters) only show/hide the checkboxes in the control panels. But then we should change the names of the Properties and query parameters. Probably something like `showGuidesCheckboxProperty` and `show2FPointsCheckboxProperty`, with query parameters `showGuidesCheckbox` and `show2FPointsCheckbox`. (Hmmm... I like those names quite a bit more.) 

It's not totally clear if this proposal would do what I proposed above. If so, I like this direction. It also has the added benefit of matching the "Show "2F Points" checkbox" wording in the Options dialog.

**Comment by arouinfar** [comment]
Thanks @pixelzoom. That all sounds reasonable and the behavior looks good in Studio.

> Visibility of the "Show 2F Points" checkbox is controlled via `add2FPointsCheckboxProperty`.  You cannot control the visiblity of these checkboxes directly or independently for each screen - that is contrary to having a global control in the Options dialog.

That makes sense to me, but it might not be clear to a client how to hide the checkbox, especially if they are starting in GOB. Is it possible to add a linked property to `global.view.options.add2FPointsCheckboxProperty` under `view.controls.controlPanel.checkboxGroup.twoFPointsCheckbox`?

**Comment by pixelzoom** [comment]
The '2F Points' checkbox is created as part of a VerticalCheckboxGroup, and I can't actually get at the actual checkbox.  I'd have to do something "clever", like lookup the checkbox by it's phetioID.

I also noticed that I was creating the 'Focal Point (F)' and '2F Points' checkboxes in the Mirror screen of GO:Basics, where they will never be relevant (because that screen only has a flat mirror, with infinite focal length.) 

So... Here's what I did:

* 'Focal Point (F)' and '2F Points' checkboxes are no longer created for the Mirror screen in GO:Basics.
* `twoFPointsCheckbox.visibleProperty` now links to `global.view.options.add2FPointsCheckboxProperty`, as shown in the Studio screenshot below.

<img width="831" alt="screenshot_1611" src="https://user-images.githubusercontent.com/3046552/158893479-b5a22a69-1435-4a98-900c-380f9680678c.png">
 
@arouinfar if this is acceptable, feel free to close.
